pkg/listwatch/listwatch.go

```diff
@@ -170,8 +173,8 @@ func (mlw multiListerWatcher) Watch(options metav1.ListOptions) (watch.Interface
 	// Allow resource versions to be "".
 	if options.ResourceVersion != "" {
 		rvs := strings.Split(options.ResourceVersion, "/")
-		if len(rvs) != len(mlw) {
-			return nil, fmt.Errorf("expected resource version to have %d parts to match the number of ListerWatchers", len(mlw))
+		if len(rvs) != 1 {
+			return nil, fmt.Errorf("expected resource version to have 1 parts to match the number of ListerWatchers, got %d", len(rvs))
```

Comment:
I don't understand the logic here. The reason we combine multiple resource versions with a slash is to later be able to distribute the resources to multiple watchers. Yet here we are enforcing that there should only be one version. So why not simply not join versions in the first place rather than fail if we did join then?

Reply:
Yes you are right but the new exception is that we always must have a single resource version (I realize this makes use of implemention details of resourceVersion right now, but until refactoring its reliable enough judging by our tests and manual testing) there is certainly room for refactoring but we would end up refactoring or removing the entire thing, the aim is to do the minimal to fix the bug yet not modify as this is aimed at 0.40 release. See https://github.com/coreos/prometheus-operator/issues/3218 for background of this specific one.